src/kpi.js

```diff
@@ -0,0 +1,46 @@
+import axios from 'axios';
+import { URL } from 'universal-url';
+
+export default function Kpi(kpiToken) {
+  const MAX_ATTEMPTS = 15 * 60; // 15 min
+
+  const delay = () => new Promise(res => setTimeout(res, 1000));
+
+  async function retryGet(url, attempt = 1) {
+    const res = await axios({
+      method: 'GET',
+      url,
+    });
+    const ifNoneMatch = new URL(url).searchParams.get('If-None-Match');
+    const ETag = res.headers.etag;
+    if (ETag === ifNoneMatch) {
+      if (attempt >= MAX_ATTEMPTS) {
+        return { data: [] };
```

Comment:
Goncalo pointed out to me that this should throw - because it's an exception: the kpi didn't load.